src/java/arjdbc/postgresql/PostgreSQLRubyJdbcConnection.java

```diff
@@ -374,12 +374,26 @@ protected void setDateParameter(final ThreadContext context,
         final int index, IRubyObject value,
         final IRubyObject attribute, final int type) throws SQLException {
 
+        if ( value instanceof RubyFloat ) {
+            final double doubleValue = ( (RubyFloat) value ).getValue();
+            if ( Double.isInfinite(doubleValue) ) {
+                setTimestampInfinity(statement, index, doubleValue);
+                return;
+            }
+        }
+
         if ( ! "Date".equals(value.getMetaClass().getName()) && value.respondsTo("to_date") ) {
             value = value.callMethod(context, "to_date");
         }
 
-        // NOTE: assuming Date#to_s does right ...
-        statement.setDate(index, Date.valueOf(value.toString()));
+        int year = RubyNumeric.num2int(value.callMethod(context, "year"));
```

Comment:
yeah, but how to get the year/month/day values otherwise with JRuby 9.1 (w/o native Date class). I mean other from parsing the string? At this point I worry more about correctness with this whole gregorian/julian/iso chronology and year 0 and BC stuff..

Reply:
the native (joda) `DateTime` could get extracted from an i-var ... but we can do that later.
could you try running some of the `bench/xxx` were this is used, just so we know if it matters at all.
would be nice to have a report here.

Reply:
Ok, I'll try to run some bench stuff on the weekend or so...meanwhile, do you have a comment about the rest of the date handling?

I just retried with the jruby-head and the 9.2 problems I pointed out above seem to be gone. So that's good news. Also, I might split the patch into two. 1. the infinity handling, 2. the year 0, BC handling stuff.

Reply:
OK, thanks- if not I will re-run some myself. we can always do the `@dt` extraction later.

> Also, I might split the patch into two. 1. the infinity handling, 2. the year 0, BC handling stuff.

fine by me as is.

> meanwhile, do you have a comment about the rest of the date handling?

Date seems to use GJ by default and we never actually use ISO in 9.2's impl so this makes sense.
it was probably wrong for me to use ISO in the first place, as it would be a source of incompatibilities ...
we just need to verify no date/time related regression happens in CI - which I expect to be an easy check.